ui/admin-portal/src/app/services/candidate-field.service.ts

```diff
@@ -406,15 +410,23 @@ export class CandidateFieldService {
    * search.
    */
   public getAddedByPartner(candidate: Candidate, source: CandidateSource): string {
-    const candidateOppForThisList: CandidateOpportunity = this.findRelevantCandidateOpp(candidate, source);
-    return candidateOppForThisList.createdBy.partner.name;
+    const candidateOppForThisList: CandidateOpportunity =
+      this.findRelevantCandidateOpp(candidate, source);
+
+    return candidateOppForThisList === null ? '?' : candidateOppForThisList.createdBy.partner.name;
   }
 
   // When displaying a submission list, will find the relevant candidate opp for given candidate.
   private findRelevantCandidateOpp(candidate: Candidate, source: CandidateSource): CandidateOpportunity {
-    return candidate.candidateOpportunities.find(
+    const opp = candidate.candidateOpportunities.find(
       opp => opp.jobOpp.submissionList.id == source.id
     );
+    if (opp === null) {
```

Comment:
Just a note that you can use simple == with null. That has the advantage of working for both null and undefined.

As it is now an undefined opp will break your code.

Reply:
Great catch! Amended and resubmitted 👍